activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb

```diff
@@ -290,9 +290,28 @@ def reset_pk_sequence!(table, pk = nil, sequence = nil) #:nodoc:
 
           if pk && sequence
             quoted_sequence = quote_table_name(sequence)
+            max_pk = select_value("select MAX(#{quote_column_name pk}) from #{quote_table_name(table)}")
+            if max_pk.nil?
+              if postgresql_version >= 100000
+                if sequence.is_a? String
+                  if sequence.include?(".")
+                    schema_name, sequence_name = sequence.split(".")
+                  else
+                    schema_name = "public"
+                    sequence_name = sequence
+                  end
+                else
+                  schema_name = sequence.schema
+                  sequence_name = sequence.identifier
+                end
+                minvalue = select_value("SELECT min_value FROM pg_sequences WHERE schemaname = '#{schema_name}' AND sequencename = '#{sequence_name}'")
```

Comment:
Let's use `pg_sequence` instead. Then we should be able to use `seqrelid = '#{quoted_sequence}'::regclass`, and the schema details will be handled for us.